stdlib/Profile/src/Profile.jl

```diff
@@ -924,16 +937,20 @@ function tree_format(frames::Vector{<:StackFrameTree}, level::Int, cols::Int, ma
                 else
                     fname = string(li.func)
                 end
-                filename = short_path(li.file, filenamemap)
+                pkgname, filename = short_path(li.file, filenamemap)
                 if showpointer
                     fname = string(
                         "0x",
                         string(li.pointer, base = 16, pad = 2*sizeof(Ptr{Cvoid})),
                         " ",
                         fname)
                 end
+                pkgcolor = get!(() -> popfirst!(Base.STACKTRACE_MODULECOLORS), PACKAGE_FIXEDCOLORS, pkgname)
+                remaining_path = rtruncto(filename, widthfile - length(pkgname) - 1)
                 strs[i] = string(stroverhead, "╎", base, strcount, " ",
-                    rtruncto(filename, widthfile),
+                    sprint((io,x) -> printstyled(io, x; color=pkgcolor), pkgname; context=io),
+                    !isempty(pkgname) && !startswith(remaining_path, "/") ? "/" : "",
+                    remaining_path,
                     ":",
                     li.line == -1 ? "?" : string(li.line),
                     "; ",
```

Comment:
Ok, GitHub doesn't seem to let me comment on lines more than three lines away from the diff changes, but this is actually pursuant to line 945 (962 with your patch):

```julia
strs[i] = ltruncto(strs[i], cols)
```

There are two problems here, one existing, one introduced.

Firstly, the existing problem: `ltruncto` / `rtruncto` mix up indices and `textwidth` and that means that `textwidth(ltruncto(str, n)) <= n` doesn't always hold. For example `textwidth(ltruncto("🍕" ^ 15))` is `19`.

Now the new problem: by introducing `sprint` + `printstyled` above, `strs[i]` now contains ANSI escape codes that are counted by `length`/`textwidth` but don't actually increase the width of the string. This is going to cause lines to be systematically over-shortened.

```julia-repl
julia> textwidth(sprint(io -> printstyled(io, "hey"), context=stdout))
6
```